src/templates/import.js

```diff
@@ -233,6 +256,22 @@ module.exports = (router) => {
       changes = true;
     }
 
+    // Check if an existing resource exists in the new project.
```

Comment:
I think I need a little more explanation for this change. Is the idea here that entity.resource will always be a string value here (i.e it's been successfully dereferenced)? If so, that makes sense, but then why check for `!changes`? Feel free to respond in dev support, it's probably easier to just discuss.

Reply:
The mentioned has less to do with the de-ref and more to do with checking if a like named resource exists in the target project even if the resource is not part of the imported template.

When the template is import and their and no resources in the export template the import will check to see if liked named resource exists. In this case the initial `entity.resource` **comes in as "customer"** as my source project contained a Customer resource.

![image](https://github.com/formio/formio/assets/112976114/73e874b4-8960-442a-a16e-914ebe07e5ad)

Over the lifecycle of the import `componentMachineNameToId` and `resourceMachineNameToId` are passed through several in various contexts relating importing roles, actions, forms, and resources.

Each of the above "items" are passed in an install method where "transform" gets called on each item of the items passed to install.

https://github.com/formio/formio/blob/0f6cafef6304b749d171884645756250763d16a9/src/templates/import.js#L1074-L1075
https://github.com/formio/formio/blob/0f6cafef6304b749d171884645756250763d16a9/src/templates/import.js#L985-L987

In the context of a resource we will arrive to this transform on the resource object.

https://github.com/formio/formio/blob/0f6cafef6304b749d171884645756250763d16a9/src/templates/import.js#L475

This function handles the def-ref but internal of this function resides `resourceMachineNameToId` which is called several times and this method is what is responsible for checking if like named resource exists in target project and if so assigns the objectId found resource using entity.resource should in the first image on this comment.

Each time `resourceMachineNameToId` its check if attributes inside the template need to be updated or if they've already been updated.

To be continued......